### PR TITLE
BOM-1300

### PR DESCRIPTION
**Pull request by awais786**
Switching from nose to pytest.
https://openedx.atlassian.net/browse/BOM-1300

### CODE REVIEW

**Comment by awais786** [comment]
```
designer/urls.py                                                         15      0      0      0   100%
-----------------------------------------------------------------------------------------------------------------
TOTAL                                                                   856     63    220     19    91%
Coverage XML written to file coverage.xml
====================== 24 passed, 4370 warnings in 13.59s ======================
The command "make test" exited with 0.
````

File: Makefile [hunk 1]
```diff
@@ -60,8 +60,7 @@ production-requirements:
 	pip install -qr requirements.txt --exists-action w
 
 test: clean test_requirements
-	coverage run ./manage.py test designer --settings=designer.settings.test
-	coverage report
+	pytest
```
Comment (jmbowman) [inline Makefile]: Not that we don't already have enough warnings to fix here, but using `python -Wd -m pytest` instead might turn up a few more.

File: requirements/dev.txt [hunk 2]
```diff
@@ -106,10 +109,11 @@ unidecode==1.1.1
 uritemplate==3.0.1
 urllib3==1.25.8
 wagtail==1.13.4
+wcwidth==0.1.8
 webencodings==0.5.1
 willow==1.0
 wrapt==1.12.0
-zipp==1.2.0               # via importlib-metadata
+zipp==3.0.0
```
Comment (jmbowman) [inline requirements/dev.txt]: zipp 2.0.0 dropped Python 3.5 support.

File: Makefile [hunk 3]
```diff
@@ -60,8 +60,7 @@ production-requirements:
 	pip install -qr requirements.txt --exists-action w
 
 test: clean test_requirements
-	coverage run ./manage.py test designer --settings=designer.settings.test
-	coverage report
+	python -Wd -m pytest
```
Comment (Ayub-Khan) [inline Makefile]: coverage ?

File: pytest.ini [hunk 4]
```diff
@@ -0,0 +1,4 @@
+[pytest]
+addopts = --ds=designer.settings.test --cov designer --cov-report term-missing --cov-report xml
```
Comment (Ayub-Khan) [inline pytest.ini]: nit*
DJANGO_SETTINGS_MODULE = designer.settings.test

**Pull request merged**